sdk/python/feast/entity.py

```diff
@@ -79,6 +80,13 @@ def __init__(
             ValueError: Parameters are specified incorrectly.
         """
         self.name = name
+        if value_type is None:
+            warnings.warn(
+                "Entity value_type will be mandatory in the next release. "
+                "Please specify a value_type for entity '%s'." % name,
+                DeprecationWarning,
```

Comment:
why not FutureWarning instead?
https://docs.python.org/3/library/exceptions.html#FutureWarning

Reply:
DeprecationWarning:
>Base class for warnings about deprecated features when those warnings are intended for other Python developers.
>
>Ignored by the default warning filters, except in the __main__ module ([PEP 565](https://peps.python.org/pep-0565/)). Enabling the [Python Development Mode](https://docs.python.org/3/library/devmode.html#devmode) shows this warning.

FutureWarning
>Base class for warnings about deprecated features when those warnings are intended for end users of applications that are written in Python.

I think DeprecationWarning is a reasonable choice but I could see both ways. Devin chose the former. 🤷

Reply:
I can’t see any deprecation actually, but if Devin will become the main maintainer, that’s fine 🤷🏻‍♂️

Reply:
I'm going to force it next release.